src/main/java/org/folio/linked/data/validation/dto/LccnPatternValidator.java

```diff
@@ -0,0 +1,38 @@
+package org.folio.linked.data.validation.dto;
+
+import static org.apache.commons.collections4.CollectionUtils.isEmpty;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.regex.Pattern;
+import org.folio.linked.data.domain.dto.LccnRequest;
+import org.folio.linked.data.validation.LccnPatternConstraint;
+
+public class LccnPatternValidator implements ConstraintValidator<LccnPatternConstraint, LccnRequest> {
+
+  private static final Pattern LCCN_STRUCTURE_A_PATTERN = Pattern.compile("( {3}|[a-z][ |a-z]{2})\\d{8} ");
+  private static final Pattern LCCN_STRUCTURE_B_PATTERN = Pattern.compile("( {2}|[a-z][ |a-z])\\d{10}");
+
+  @Override
+  public boolean isValid(LccnRequest lccnRequest, ConstraintValidatorContext constraintValidatorContext) {
+    if (isCurrent(lccnRequest)) {
+      return lccnRequest.getValue()
+        .stream()
+        .anyMatch(this::hasValidPattern);
```

Comment:
it was noneMatch(this::hasInvalidPattern), is anyMatch(this::hasValidPattern) expected here?

Reply:
sure